src/v2/DevTools/mockLocation.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Writable window.location for use in Jest tests
+ */
+export const mockLocation = (options: Partial<Location> = {}) => {
```

Comment:
Awesome this will simplify things when we try to upgrade jest in the future. Jest >=25 uses a version of jsdom that doesn't allow mocking `window.location` and have been a blocker on all my attempts to upgrade, having one common location will hopefully make this easier.